cmd_migrate_db.go

```diff
@@ -0,0 +1,705 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	_ "net/http/pprof" // Register pprof handlers
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/btcsuite/btclog/v2"
+	"github.com/btcsuite/btcwallet/walletdb"
+	"github.com/jessevdk/go-flags"
+	"github.com/lightninglabs/lndinit/migratekvdb"
+	"github.com/lightningnetwork/lnd/channeldb"
+	"github.com/lightningnetwork/lnd/kvdb"
+	"github.com/lightningnetwork/lnd/kvdb/postgres"
+	"github.com/lightningnetwork/lnd/kvdb/sqlbase"
+	"github.com/lightningnetwork/lnd/kvdb/sqlite"
+	"github.com/lightningnetwork/lnd/lncfg"
+	"github.com/lightningnetwork/lnd/watchtower/wtdb"
+)
+
+var (
+	// alreadyMigratedKey is the key under which we add a tag in the target/
+	// destination DB after we've successfully and completely migrated it
+	// from a source DB.
+	alreadyMigratedKey = []byte("data-migration-already-migrated")
+
+	// defaultDataDir is the default data directory for lnd.
+	defaultDataDir = filepath.Join(btcutil.AppDataDir("lnd", false), "data")
+)
+
+const (
+	// walletMetaBucket is the name of the meta bucket in the wallet db
+	// for the wallet ready marker.
+	walletMetaBucket = "lnwallet"
```

Comment:
Do we need this for all the other sub DBs?

Reply:
no just the walletdb see also:

https://github.com/lightningnetwork/lnd/blob/a7c2c47397258eac69e88d232fbf3ccc503ad308/lncfg/db.go#L382

and in the same file for the other cases.